apps/backoffice-v2/src/common/components/molecules/DocumentTracker/hooks/useDocumentTracker.tsx

```diff
@@ -0,0 +1,84 @@
+import { useState } from 'react';
+import { titleCase } from 'string-ts';
+
+import { useRequestDocumentsMutation } from '@/domains/documents/hooks/mutations/useRequestDocumentsMutation';
+import { TrackedDocument } from '@/domains/documents/hooks/schemas/document';
+import { TWorkflowDefinitionById } from '@/domains/workflow-definitions/fetchers';
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { documentStatusToIcon, Icon } from '../constants';
+import { useDocumentsTrackerItemsQuery } from '@/domains/documents/hooks/queries/useDocumentsTrackerItemsQuery';
+
+export interface IUseDocumentTrackerLogicParams {
+  plugins: Array<
+    | NonNullable<NonNullable<TWorkflowDefinitionById['extensions']>['apiPlugins']>[number]
+    | NonNullable<
+        NonNullable<TWorkflowDefinitionById['extensions']>['childWorkflowPlugins']
+      >[number]
+    | NonNullable<NonNullable<TWorkflowDefinitionById['extensions']>['commonPlugins']>[number]
+  >;
+  workflow: TWorkflowById;
+}
+
+export const useDocumentTracker = ({ plugins, workflow }: IUseDocumentTrackerLogicParams) => {
+  const { data: documents, isLoading: isLoadingDocuments } = useDocumentsTrackerItemsQuery({
+    workflowId: workflow.id,
+  });
+
+  const [open, onOpenChange] = useState(false);
+  const [selectedIdsToRequest, setSelectedIdsToRequest] = useState<string[]>([]);
+
+  const { mutate: requestDocuments } = useRequestDocumentsMutation({
+    onSuccess: () => {
+      setSelectedIdsToRequest([]);
+      onOpenChange(false);
+    },
+  });
+
+  const onRequestDocuments = () => requestDocuments({ documentIds: selectedIdsToRequest });
+
+  const getSubItems = (doc: TrackedDocument) => {
+    const { documentId } = doc;
+
+    if (!documentId) {
+      return {
+        leftIcon: null,
+        text: 'Document ID is missing',
+        itemClassName: 'p-1',
+      };
+    }
+
+    return {
+      leftIcon: (
+        <div
```

Comment:
Please use our button component or the native HTML button element for clickable UI at least of similar nature. 🙏🏼